### ISSUE TITLE
The proposed preference attention layer does not seem necessary

### ISSUE THREAD

**Issue by dssrgu** [csmile-1006/PreferenceTransformer#3]
Hi,

Thanks for sharing the code!

I have a few questions regarding how to reproduce the results in the paper.

1. README says that you should run the following command to train a preference transformer model:
```python
# Preference Transfomer (PT)
CUDA_VISIBLE_DEVICES=0 python -m JaxPref.new_preference_reward_main --use_human_label True --comment {experiment_name} --transformer.embd_dim 256 --transformer.n_layer 1 --transformer.n_head 4 --env {D4RL env name} --logging.output_dir './logs/pref_reward' --batch_size 256 --num_query {number of query} --query_len 100 --n_epochs 10000 --skip_flag 0 --seed {seed} --model_type PrefTransformer
```

However, I notice that this command will set `config.used_weighted_sum` in https://github.com/csmile-1006/PreferenceTransformer/blob/8b34c6979fd08522cd28a3e5d7ffac132f341790/JaxPref/PrefTransformer.py#L36
to `False`, leading to **not** using the preference attention layer at all.
https://github.com/csmile-1006/PreferenceTransformer/blob/8b34c6979fd08522cd28a3e5d7ffac132f341790/flaxmodels/flaxmodels/gpt2/trajectory_gpt2.py#L403-L408

Is it correct that you do not need the `--transformer.use_weighted_sum` flag?

2. I tried to reproduce the paper's results both with and without the `--transformer.use_weighted_sum` flag. Other than this flag, I strictly followed the guidelines in the README. In detail, when training the reward model, I set `--num_query` to 500 for *-medium-replay datasets and 100 for *-medium-expert datasets. When running IQL with the learned reward model, I set `--seq_len=100`, `--eval_Interval=5000`, `--config=configs/mujoco_config.py`, and `--eval_episodes=10`. Below are the IQL results for 8 seeds (0~7):

| use_weighted_sum | hopper-medium-replay-v2 | hopper-medium-expert-v2 | walker2d-medium-replay-v2 | walker2d-medium-expert-v2 |
|------------------|-------------------------|-------------------------|---------------------------|---------------------------|
| False            |           70.03 (24.06) |           87.31 (13.15) |              75.82 (2.37) |             109.93 (0.83) |
| True             |           68.83 (23.34) |           68.54 (32.64) |              76.48 (3.30) |             109.78 (0.47) |

(Values in the parentheses denote the std of the normalized return.)

Surprisingly, I find that **not** using the preference attention layer actually performs better. Does this mean that the preference attention layer is not helpful? Or am I missing something? Also, do you have any ablation results regarding each component of your method?

In addition, I failed to reproduce the results on hopper-medium-replay (84.54) both with and without the preference attention layer. Can you take a look at this issue?

**Comment by csmile-1006** [comment]
Hello!

1. I believe I may have missed some details in the README. It is correct to set the "--transformer.use_weighted_sum" flag to True.

2. The evaluation performance is highly sensitive to the seed number, particularly in the Mujoco locomotion environments. For our experiments, we utilized seed numbers starting from 3407. However, it's possible that the code may not be able to reproduce the results in the paper precisely due to various factors such as the version of JAX/Tensorflow libraries, type of GPUs, etc. The gain from using the preference attention layer is comparatively marginal in Mujoco locomotion environments as the task and reward function is relatively Markovian when compared to Antmaze navigation tasks.

**Comment by dssrgu** [comment]
Hi, I additionally ran the Antmaze experiments since you suggested the performance gain would be more pronounced in Antmaze:

| use_weighted_sum | antmaze-medium-play-v2 | antmaze-medium-diverse-v2 | antmaze-large-play-v2 | antmaze-large-diverse-v2 |
|------------------|------------------------|---------------------------|-----------------------|--------------------------|
| False            |           74.00 (4.56) |              65.25 (6.23) |          51.00 (9.55) |             16.88 (8.56) |
| True             |           66.50 (5.76) |              66.63 (6.30) |         34.88 (13.28) |            21.50 (15.80) |

It still seems the performance is better off without the preference attention layer.

I understand that the evaluation performance of RL can be sensitive to random seeds (or other subtle implementation details), but I am doubtful whether the preference attention layer consistently having no performance gain on all of the 8 tasks can solely be attributed to those factors.

Thank you for the response.
